src/app/ubs-admin/components/ubs-admin-employee/employee-form/employee-form.component.spec.ts

```diff
@@ -0,0 +1,175 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+
+import { RouterTestingModule } from '@angular/router/testing';
+import { HttpClientTestingModule } from '@angular/common/http/testing';
+import { MatDialogRef, MatDialogModule, MAT_DIALOG_DATA } from '@angular/material/dialog';
+
+import { EmployeeFormComponent } from './employee-form.component';
+import { ReactiveFormsModule } from '@angular/forms';
+import { TranslateModule } from '@ngx-translate/core';
+
+describe('EmployeeFormComponent', () => {
+  let component: EmployeeFormComponent;
+  let fixture: ComponentFixture<EmployeeFormComponent>;
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      declarations: [EmployeeFormComponent],
+      imports: [MatDialogModule, HttpClientTestingModule, RouterTestingModule, ReactiveFormsModule, TranslateModule.forRoot()],
+      providers: [
+        { provide: MatDialogRef, useValue: {} },
+        { provide: MAT_DIALOG_DATA, useValue: {} }
+      ]
+    }).compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(EmployeeFormComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('Employee form should create', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('Employee form should be invalid when empty', () => {
+    expect(component.employeeForm.valid).toBeFalsy();
+  });
+
+  it('First name field should be invalid when empty', () => {
+    const firstName = component.employeeForm.controls.firstName;
+    expect(firstName.valid).toBeFalsy();
+  });
```

Comment:
In both tests, you should create a situation when this form is empty
Somebody could add some default values for this form and your tests will fail, just because you haven't created the right conditions for that test
So you should specify right in the test that fields are empty
In the first test the whole form should be empty but in the second only the First name field (!but other fields should be fulfilled to show that the form will be invalid even we have only the First name field empty )

Reply:
And the same it other tests where you check the form with empty fields to be invalid

Reply:
Done